plugins/source/postgresql/client/cdc.go

```diff
@@ -0,0 +1,290 @@
+package client
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/jackc/pglogrepl"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgproto3"
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+func (c *Client) tablesWithPks() []string {
+	var tables []string
+	for _, table := range c.Tables {
+		if len(table.PrimaryKeys()) > 0 {
+			tables = append(tables, table.Name)
+		}
+	}
+	return tables
+}
+
+func (c *Client) createPublicationForTables(ctx context.Context, conn *pgconn.PgConn) error {
+	tables := c.tablesWithPks()
+	sql := fmt.Sprintf("CREATE PUBLICATION %s FOR TABLE %s", pgx.Identifier{c.spec.Name}.Sanitize(), strings.Join(tables, ","))
+	reader := conn.Exec(ctx, sql)
+	if _, err := reader.ReadAll(); err != nil {
+		var pgErr *pgconn.PgError
+		if !errors.As(err, &pgErr) {
+			// not recoverable error
+			return fmt.Errorf("failed to create publication: %w", err)
+		}
+		// 42710 means publication already exists
+		if pgErr.Code != "42710" {
+			// not recoverable error
+			return fmt.Errorf("failed to create publication with pgerror %s: %w", pgErrToStr(pgErr), err)
+		}
+		sql = fmt.Sprintf("ALTER PUBLICATION %s SET TABLE %s", pgx.Identifier{c.spec.Name}.Sanitize(), strings.Join(tables, ","))
+		reader := conn.Exec(ctx, sql)
+		if _, err := reader.ReadAll(); err != nil {
+			return fmt.Errorf("failed to alter publication: %w", err)
+		}
+	}
+	return nil
+}
+
+func (c *Client) startCDC(ctx context.Context, conn *pgconn.PgConn) (string, error) {
+	var err error
```

Comment:
This line is unnecessary with all the `:=` combined definition-assignments below.